src/main/java/us/kbase/narrativemethodstore/db/mongo/MongoDynamicRepoDB.java

```diff
@@ -191,23 +194,27 @@ public void registerRepo(String userId, RepoProvider repoDetails)
                         "] -> [" + newUrl + "]");
         }
         final RepoData repoData = JsonRepoProvider.repoProviderToData(this, repoDetails);
-        final DBCollection hist = db.getCollection(TABLE_REPO_HISTORY);
-        hist.insert(new BasicDBObject(FIELD_RH_MODULE_NAME, repoModuleName)
+        final MongoCollection<Document> hist = db.getCollection(TABLE_REPO_HISTORY);
+        hist.insertOne(new Document()
+                .append(FIELD_RH_MODULE_NAME, repoModuleName)
                 .append(FIELD_RH_VERSION, newVersion)
-                .append(FIELD_RH_REPO_DATA, toDBObject(repoData)));
+                .append(FIELD_RH_REPO_DATA, toDocument(repoData)));
         
-        final DBCollection data = db.getCollection(TABLE_REPO_INFO);
+        final MongoCollection<Document> data = db.getCollection(TABLE_REPO_INFO);
         //should just do an upsert rather than handling the logic application side
         if (wasReg) {
-            final Map<String, Object> info = toMap(data.findOne(
-                    new BasicDBObject(FIELD_RI_MODULE_NAME, repoModuleName)));
-            info.put(FIELD_RI_LAST_VERSION, newVersion);
-            info.put(FIELD_RI_STATE, RepoState.ready.toString());
-            // this is race condition city
-            data.update(new BasicDBObject(FIELD_RI_MODULE_NAME, repoModuleName),
-                    new BasicDBObject("$set", info));
+            Document existingInfo = data.find(Filters.eq(FIELD_RI_MODULE_NAME, repoModuleName)).first();
+
+            if (existingInfo != null) {
+                existingInfo.put(FIELD_RI_LAST_VERSION, newVersion);
+                existingInfo.put(FIELD_RI_STATE, RepoState.ready.toString());
+
+                // update the existing document
+                data.replaceOne(Filters.eq(FIELD_RI_MODULE_NAME, repoModuleName), existingInfo);
```

Comment:
Even though it seems like it's not necessary, let's keep using the update / $set approach just in case

Reply:
👍

Reply:
:+1: